pytorch_lightning/trainer/properties.py

```diff
@@ -420,16 +420,16 @@ def __setstate__(self, state):
         self.__dict__ = state
 
     @property
-    def require_distributed_sampler(self):
-        if self.accelerator_backend is not None:
-            return self.accelerator_backend.require_distributed_sampler
+    def require_distributed_sampler(self) -> bool:
+        if self.accelerator is not None:
+            return self.accelerator.require_distributed_sampler
         return self._distrib_type in (
             DistributedType.HOROVOD, DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2
         ) or self._device_type == DeviceType.TPU
```

Comment:
is this required in any case now?
```suggestion
```

Reply:
done here #5975 
this pr is just for typing